services/apps/alcs/src/alcs/planning-review/planning-review-decision/planning-review-decision.service.ts

```diff
@@ -0,0 +1,351 @@
+      );
+    }
+
+    const planningReview = await this.planningReviewService.getByFileNumber(
+      createDto.planningReviewFileNumber,
+    );
+
+    const decision = new PlanningReviewDecision({
+      resolutionYear: new Date().getFullYear(),
+      planningReviewUuid: planningReview.uuid,
+    });
+
+    const savedDecision = await this.planningReviewDecisionRepository.save(
+      decision,
+      {
+        transaction: true,
+      },
+    );
+
+    return this.get(savedDecision.uuid);
+  }
+
+  async delete(uuid) {
+    const planningReviewDecision =
+      await this.planningReviewDecisionRepository.findOne({
+        where: { uuid },
+        relations: {
+          outcome: true,
+          documents: {
+            document: true,
+          },
+          planningReview: true,
+        },
+      });
+
+    if (!planningReviewDecision) {
+      throw new ServiceNotFoundException(
+        `Failed to find decision with uuid ${uuid}`,
+      );
+    }
+
+    for (const document of planningReviewDecision.documents) {
+      await this.documentService.softRemove(document.document);
+    }
+    await this.planningReviewDecisionRepository.save(planningReviewDecision);
+
+    await this.planningReviewDecisionRepository.softRemove([
+      planningReviewDecision,
+    ]);
+    await this.updateDecisionDates(planningReviewDecision);
+  }
+
+  async attachDocument(decisionUuid: string, file: MultipartFile, user: User) {
+    const decision = await this.getOrFail(decisionUuid);
+    const document = await this.documentService.create(
+      `decision/${decision.uuid}`,
+      file.filename,
+      file,
+      user,
+      DOCUMENT_SOURCE.ALC,
+      DOCUMENT_SYSTEM.ALCS,
+    );
+    const appDocument = new PlanningReviewDecisionDocument({
+      decision,
+      document,
+    });
+
+    return this.decisionDocumentRepository.save(appDocument);
+  }
+
+  async deleteDocument(decisionDocumentUuid: string) {
+    const decisionDocument =
+      await this.getDecisionDocumentOrFail(decisionDocumentUuid);
+
+    await this.decisionDocumentRepository.softRemove(decisionDocument);
+    return decisionDocument;
+  }
+
+  async getDownloadUrl(decisionDocumentUuid: string, openInline = false) {
+    const decisionDocument =
+      await this.getDecisionDocumentOrFail(decisionDocumentUuid);
+
+    return this.documentService.getDownloadUrl(
+      decisionDocument.document,
+      openInline,
+    );
+  }
+
+  async getDownloadForPortal(decisionDocumentUuid: string) {
+    const decisionDocument = await this.decisionDocumentRepository.findOne({
+      where: {
+        decision: {
+          isDraft: false,
+        },
+        uuid: decisionDocumentUuid,
+      },
+      relations: {
+        document: true,
+      },
+    });
+
+    if (decisionDocument) {
+      return this.documentService.getDownloadUrl(
+        decisionDocument.document,
+        true, // FIXME: Document does not open inline despite flag being true
```

Comment:
is this still relevant after changes that Sandra made with iframe?

Reply:
Sandra has not applied the file changes to ALCS yet, only the portal, so I think it applies for now